mathics/builtin/patterns/composite.py

```diff
@@ -0,0 +1,516 @@
+            'Infix[{symbol, pattern}, ":", 150, Left]'
+        )
+    }
+    summary_text = "a named pattern"
+
+    def init(
+        self, expr: Expression, evaluation: OptionalType[Evaluation] = None
+    ) -> None:
+        if len(expr.elements) != 2:
+            self.error("patvar", expr)
+        varname = expr.elements[0].get_name()
+        if varname is None or varname == "":
+            self.error("patvar", expr)
+        super().init(expr, evaluation=evaluation)
+        self.varname = varname
+        self.pattern = BasePattern.create(expr.elements[1], evaluation=evaluation)
+
+    def __repr__(self):
+        return "<Pattern: %s>" % repr(self.pattern)
+
+    def get_match_count(
+        self, vars_dict: OptionalType[dict] = None
+    ) -> Union[int, tuple]:
+        return self.pattern.get_match_count(vars_dict)
+
+    def match(self, expression: Expression, pattern_context: dict):
+        """Match with a (named) pattern"""
+        from mathics.builtin.patterns.defaults import OptionsPattern
+
+        yield_func = pattern_context["yield_func"]
+        vars_dict = pattern_context["vars_dict"]
+
+        existing = vars_dict.get(self.varname, None)
+        if existing is None:
+            new_vars_dict = vars_dict.copy()
+            new_vars_dict[self.varname] = expression
+            pattern_context = pattern_context.copy()
+            pattern_context["vars_dict"] = new_vars_dict
+            # for vars_2, rest in self.pattern.match(
+            #    expression, new_vars_dict, evaluation):
+            #    yield vars_2, rest
+            if isinstance(self.pattern, OptionsPattern):
+                self.pattern.match(
+                    expression=expression, pattern_context=pattern_context
+                )
+            else:
+                self.pattern.match(
+                    expression=expression, pattern_context=pattern_context
+                )
+        else:
+            if existing.sameQ(expression):
+                yield_func(vars_dict, None)
+
+    def get_match_candidates(self, elements: tuple, pattern_context: dict) -> tuple:
+        """
+        Return a sub-tuple of elements that match with
+        the pattern.
+        Optional parameters provide information
+        about the context where the elements and the
+        patterns come from.
+        """
+        vars_dict = pattern_context.get("vars_dict", {})
+
+        existing = vars_dict.get(self.varname, None)
+        if existing is None:
+            return self.pattern.get_match_candidates(elements, pattern_context)
+
+        # Treat existing variable as verbatim
+        verbatim_expr = Expression(SymbolVerbatim, existing)
+        verbatim = Verbatim(verbatim_expr)
+        return verbatim.get_match_candidates(elements, pattern_context)
+
+
+class Repeated(PostfixOperator, PatternObject):
+    """
+    <url>:WMA link:https://reference.wolfram.com/language/ref/Repeated.html</url>
+
+    <dl>
+      <dt>'Repeated[$pattern$]'
```

Comment:
Now here we use something different from `pat` and `patt`. Pick either `pattern` or `pat` (and not `patt`) and make things consistent.

Reply:
Will be addressed in next PR